jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildImageTask.java

```diff
@@ -39,88 +38,48 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import javax.annotation.Nullable;
-import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
-import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.Internal;
-import org.gradle.api.tasks.Nested;
-import org.gradle.api.tasks.Optional;
 import org.gradle.api.tasks.TaskAction;
 
 /** Builds a container image. */
-public class BuildImageTask extends DefaultTask {
+public class BuildImageTask extends TaskConfiguration {
 
   /** Directory name for the cache. The directory will be relative to the build output directory. */
   private static final String CACHE_DIRECTORY_NAME = "jib-cache";
 
   /** {@code User-Agent} header suffix to send to the registry. */
   private static final String USER_AGENT_SUFFIX = "jib-gradle-plugin";
 
-  @Nullable private ImageConfiguration from;
-  @Nullable private ImageConfiguration to;
-  @Nullable private List<String> jvmFlags;
-  @Nullable private String mainClass;
-  private boolean reproducible;
-  @Nullable private Class<? extends BuildableManifestTemplate> format;
-  private boolean useOnlyProjectCache;
-
-  @Nested
-  @Nullable
-  public ImageConfiguration getFrom() {
-    return from;
-  }
-
-  @Nested
-  @Nullable
-  public ImageConfiguration getTo() {
-    return to;
-  }
-
-  @Input
-  @Nullable
-  public List<String> getJvmFlags() {
-    return jvmFlags;
-  }
-
-  @Input
-  @Nullable
-  @Optional
-  public String getMainClass() {
-    return mainClass;
-  }
-
-  @Input
-  public boolean getReproducible() {
-    return reproducible;
-  }
-
-  @Input
+  /** Converts an {@link ImageConfiguration} to an {@link Authorization}. */
+  @Internal
```

Comment:
does gradle complain that this is a getter? I'm not sure if my memory is correct, but the `@Internal` doesn't seem required here?

Reply:
Oh actually nvm, it is actually still barfing that it is trying to find the "imageAuthorization" property.